src/fides/api/ops/service/connectors/erasure_email_connector.py

```diff
@@ -0,0 +1,201 @@
+                )
+            # synchronous for now since failure to send is considered a connection test failure
+            send_single_erasure_email(
+                db=db,
+                subject_email=self.config.test_email_address,
+                subject_name=self.config.third_party_vendor_name,
+                batch_identities=list(self.identities_for_test_email.values()),
+                test_mode=True,
+            )
+        except MessageDispatchException as exc:
+            logger.info("Email connector test failed with exception {}", exc)
+            return ConnectionTestStatus.failed
+        return ConnectionTestStatus.succeeded
+
+    def needs_email(
+        self, user_identities: Dict[str, Any], privacy_request: PrivacyRequest
+    ) -> bool:
+        """Schedules an erasure email for erasure privacy requests containing the required identities"""
+        erasure_rules: List[Rule] = privacy_request.policy.get_rules_for_action(
+            action_type=ActionType.erasure
+        )
+        return bool(
+            erasure_rules
+            and filter_user_identities_for_connector(self.config, user_identities)
+        )
+
+    def batch_email_send(self, privacy_requests: Query) -> None:
+        skipped_privacy_requests: List[str] = []
+        batched_identities: List[str] = []
+
+        for privacy_request in privacy_requests:
+            user_identities: Dict[str, Any] = privacy_request.get_cached_identity_data()
+            filtered_user_identities: Dict[
+                str, Any
+            ] = filter_user_identities_for_connector(self.config, user_identities)
+            if filtered_user_identities:
+                batched_identities.extend(filtered_user_identities.values())
+            else:
+                skipped_privacy_requests.append(privacy_request.id)
+
+        if not batched_identities:
+            logger.info(
+                "Skipping erasure email send for connector: '{}'. "
+                "No corresponding user identities found for pending privacy requests.",
+                self.configuration.name,
+            )
+            return
+
+        logger.info(
+            "Sending batched erasure email for connector {}...",
+            self.configuration.name,
+        )
+
+        db = Session.object_session(self.configuration)
+
+        try:
+            send_single_erasure_email(
+                db=db,
+                subject_email=self.config.recipient_email_address,
+                subject_name=self.config.third_party_vendor_name,
+                batch_identities=batched_identities,
+                test_mode=False,
+            )
+        except MessageDispatchException as exc:
+            logger.info("Erasure email failed with exception {}", exc)
+            raise
+
+        # create an audit event for each privacy request ID
+        for privacy_request in privacy_requests:
```

Comment:
You might add an extra check here to make sure the privacy request wasn't skipped before you add a log saying it erasure email instructions were dispatched.

Reply:
I had this for the `consent_email_connector`, updating now!